### PR TITLE
[FIX] 12.0 sale_invoice_group_method: Get correct invoice_group_method_id

### PR DESCRIPTION
**Pull request by Roodin**
Get the field from commercial partner if it is not set in partner

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Take a look to https://github.com/OCA/account-invoicing/tree/12.0/sale_order_invoicing_grouping_criteria that is less invasive.

File: sale_invoice_group_method/model/partner.py [hunk 1]
```diff
@@ -11,3 +11,9 @@ class ResPartner(models.Model):
         string='Default Invoice Group Method',
         comodel_name='sale.invoice.group.method'
     )
+
+    def _get_invoice_group_method_id(self):
+        if self.invoice_group_method_id:
+            return self.invoice_group_method_id
+        else:
```
Comment (nikul-serpentcs) [inline sale_invoice_group_method/model/partner.py:18]:
```suggestion
        return self.commercial_partner_id.invoice_group_method_id
```

File: sale_invoice_group_method/model/partner.py [hunk 2]
```diff
@@ -11,3 +11,9 @@ class ResPartner(models.Model):
...
+            return self.commercial_partner_id.invoice_group_method_id
```
Comment (nikul-serpentcs) [inline sale_invoice_group_method/model/partner.py:19]: @Roodin no need to else part

Comment (pedrobaeza) [reply sale_invoice_group_method/model/partner.py:19]: Both are needed, but a compact way to put it is:
```python
return self.invoice_group_method_id or self.commercial_partner_id.invoice_group_method_id
```

Comment (nikul-serpentcs) [reply sale_invoice_group_method/model/partner.py:19]: @pedrobaeza  I mean method should be
```
def _get_invoice_group_method_id(self):
    if self.invoice_group_method_id:
        return self.invoice_group_method_id
    return self.commercial_partner_id.invoice_group_method_id
```

Comment (pedrobaeza) [reply sale_invoice_group_method/model/partner.py:19]: And it's less good than my proposal as you can see.

**Comment by marcelsavegnago** [comment]
Is there something strange about this PR or am I wrong?

image

**Pull request closed without merging**